emily/handler/src/common/error.rs

```diff
@@ -128,21 +97,130 @@ pub enum Error {
     InconsistentState(Inconsistency),
 
     /// API is reorganizing.
-    #[error("Api is reorganizing around new chain tip {0:?}")]
+    #[error("the API is reorganizing around new chain tip {0:?}")]
     Reorganizing(Chainstate),
 
     /// An entry update version conflict in a resource update resulted
     /// in an update not being performed.
-    #[error("Version conflict")]
-    VersionConflict,
-
-    /// Bad request
-    #[error("Bad request {0}")]
-    BadRequest(String),
+    #[error("there was a conflict when attempting to update the database; {0}")]
+    VersionConflict(#[source] Box<ConditionalCheckFailedException>),
 
     /// Deserialization error
     #[error("Deserialization error: {0}")]
     Deserialization(String),
+
+    /// This happens if the deposit entry that was stored in the database
+    /// was invalid, or if the deposit entry that we are creating to store
+    /// in the database is invalid.
+    #[error("deposit entry failed validation; {0}; ID: {1}")]
+    DepositEntry(&'static str, DepositEntryKey),
+
+    /// This happens when there is a mismatch in the outpoint of the new
+    /// deposit event and the fetched deposit entry. Seeing this is
+    /// probably due to a programming error.
+    #[error("mismatch when updateing deposit request; existing: {0}; update: {1}")]
+    DepositUpdate(DepositEntryKey, DepositEntryKey),
```

Comment:
Might have called this `DepositOutputMismatch` or something less generic, since the described error is pretty specific.

Reply:
Changed in https://github.com/stacks-sbtc/sbtc/pull/1627/commits/110aa6d43414fbb25618ebb3b118d157dc0d7f66